### PR TITLE
Android library build script fixes, and Android thread lock on remote disconnect.

### PR DESCRIPTION
**Pull request by wintersandroid**
## Fixes thread lock on android when remote disconnects.

Found that the client would lock when the remote disconnects, this was due to the input thread not being
told to exit.

## Android Library Build Scripts
Found that I couldn't build the android libraries "out-of-the-box". Have updated the scripts to be able to build. Also found that repeated builds (because I was fixing them) hammered github and I seemd to get rate limited, so changed the scripts to pull a zip at the required tag, and use that if needed again. Makes the process much faster.

### CODE REVIEW

File: client/Android/android_freerdp.c [hunk 1]
```diff
@@ -619,6 +619,8 @@ static int android_freerdp_run(freerdp* instance)
 
 	if (async_input && inputThread)
 	{
+    wMessageQueue* input_queue = freerdp_get_message_queue(instance, FREERDP_INPUT_MESSAGE_QUEUE);
+    MessageQueue_PostQuit(input_queue, 0);
```
Comment (akallabeth) [inline client/Android/android_freerdp.c]: Formatting, space vs tab

**Comment by akallabeth** [comment]
@freerdp-bot test

File: client/Android/Studio/freeRDPCore/build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion = 27
-    buildToolsVersion = "27.0.3"
+    compileSdkVersion = 28
+    buildToolsVersion = "28.0.3"
 
     defaultConfig {
         minSdkVersion 14
```
Comment (akallabeth) [inline client/Android/Studio/freeRDPCore/build.gradle]: Might be a good idea to raise this to `21` too

File: client/Android/android_freerdp.c (hunk 1 above)
Comment (wintersandroid) [reply client/Android/android_freerdp.c]: Ok, done...

File: client/Android/Studio/freeRDPCore/build.gradle (hunk 2 above)
Comment (wintersandroid) [reply client/Android/Studio/freeRDPCore/build.gradle]: Good spotting. Done..

**Comment by akallabeth** [comment]
@freerdp-bot test

File: client/Android/Studio/aFreeRDP/build.gradle [hunk 3]
```diff
@@ -16,7 +16,7 @@ android {
     signingConfigs {
         release {
             storeFile file(RELEASE_STORE_FILE)
-            storePassword RELEASE_STORE_PASSWORD
+            storePassword RELEASE_KEY_PASSWORD
```
Comment (akallabeth) [inline client/Android/Studio/aFreeRDP/build.gradle]: Sorry, missed this one, please leave the old name. Will break existing build setups.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Pull request merged**